### ISSUE TITLE
throw error Unhandled 'error' event

### ISSUE THREAD

**Issue by liveme** [sasstools/gulp-sass-lint#24]
The error msg:

``` javascript
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: Cannot read property 'start' of undefined
Segmentation fault: 11
```

**Comment by DanPurdy** [comment]
This is probably an issue with `sass-lint` itself, could you provide a little more information, is there any more to that error message?

**Comment by liveme** [comment]
node version is v0.12.2
when useing  `gulp.src(['src/sass/**/*.s+(a|c)ss'])`  in the gulpfile and change the sass file, it will trigger the error and interrupt the process, but with the `gulp.src(['src/sass/*.s+(a|c)ss'])` can work well.
Here is about `sass-lint` part in gulpfile:

``` javascript
gulp.src(['src/sass/**/*.s+(a|c)ss'])
            .pipe(sassLint())
            .pipe(sassLint.format())
```

Thanks!

**Comment by twlk28** [comment]
i met the same problem ：

``` zsh
  31:27  warning  Space expected after `:`                                        space-after-colon
  31:35  warning  Multiline style comments should not be used                     no-css-comments
  33:1   warning  Multiline style comments should not be used                     no-css-comments
  34:9   warning  Strings must use single quotes                                  quotes
  36:9   warning  Strings must use single quotes                                  quotes

✖ 44 problems (0 errors, 44 warnings)


events.js:141
      throw er; // Unhandled 'error' event
      ^
Error: Cannot read property 'start' of undefined
➜  del1  which node
/Users/adi/.nvm/versions/node/v4.2.4/bin/node
➜  del1  gulp -version
[13:58:26] CLI version 3.9.0
[13:58:26] Local version 3.9.0
```

## gulpfile.js

``` javascript
gulp.task('sass_lint', function(){
        return gulp.src(['src/sass/**/*.scss'])
            .pipe(sassLint())
            .pipe(sassLint.format())
            .pipe(sassLint.failOnError())
    })

gulp.watch('src/sass/**', ['sass_lint'])

```

**Comment by stones** [comment]
This happened to me, but only when I used the glob ( ie 'src/sass/*_/_.scss' ), when i gave it a full path to the folder, it worked.

**Comment by cyppher** [comment]
This might have to do with buffer size. Try increasing maxBuffer like this (it resolved the same issue for me):

```
    .pipe(sassLint({
      'maxBuffer': 1228800
    }))
    .pipe(sassLint.format())
```

> maxBuffer specifies the largest amount of data allowed on stdout or stderr - if this value is exceeded then the child process is killed.

https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback

**Comment by timschoch** [comment]
Same problem here... My intended config looks like this:

```
gulp.task('lint-scss', function () {
    gulp.src('src/sass/**/*.s+(a|c)ss')
        .pipe($.sassLint())
        .pipe($.sassLint.format());
});
```

First I tried to increase maxBuffer to a ridiculous amount `999999999999` but to no avail.

Then I removed the glob and used `gulp.src('./src/scss/core/_config.scss')` instead, this gives me a new error - no matter what the content on line 4556 or befor is.

```
4556:1  error  Please check validity of the block starting from line #4556  Fatal
```

Current config:

$ gulp -version
CLI version 0.4.0
Local version 3.9.0

$ npm gulp-scss-lint -v
3.3.12

**Comment by DanPurdy** [comment]
@timschoch that error you're seeing is a parse error so either the code you've written is invalid or more likely it's a bug in the AST we use. We're currently in the process of updating our rules to work with the new version of the AST so it may disappear shortly. You should try and isolate what code is failing and post it in the sass-lint issues, the line number in these cases can be completely off.

**Comment by NameFILIP** [comment]
Tabs vs Spaces:

Successful:

```
.barchart {
  .row {
    margin-bottom: 40px;
<space><space>}
}
```

Failing:

```
.barchart {
  .row {
    margin-bottom: 40px;
<tab>}
}
```

**Comment by DanPurdy** [comment]
Yeah this was a problem with the AST we were using, our 1.6 update should solve a lot of this while h will hopefully be out very soon!

**Comment by segovia94** [comment]
I had the same issue but it fixed once I changed the file line endings from windows CRLF to unix LF.

**Comment by alecdwm** [comment]
I had the same issue, worked around by converting indentation from tabs to spaces.

It also occurs in my main file whose' last line is an @import directive.

```
/*** Vendor **********************************************************/
@import "../vendor/responsive-nav/responsive-nav"
```

Ending the file with `//` worked around this.

**Comment by fxm90** [comment]
Anything new on this bug? The solutions from above didn't work for me :/

**Comment by Sin360** [comment]
https://github.com/sasstools/sass-lint/pull/612

Tried the fix, and it works. We now have to wait for a release for our ci builds to work :(

**Comment by DanPurdy** [comment]
These issues should be fixed now as we started a fork of the AST and fixed issues with CRLF line endings for our 1.6 and up releases.

If you see any more of these issues it would be good to try using the `sass-lint` CLI directly to see if the problem arises from there and then posting an issue with the corresponding repo.

Indentation rule will still have issues with tabs at moment but these should be fixed soon.

**Comment by brownrw8** [comment]
Just had the same problem... fixed it by running npm install on Windows CMD instead of GitBash (MINGW64).
